packages/core/src/services/ai/helpers.ts

```diff
@@ -3,15 +3,53 @@ import { createAzure } from '@ai-sdk/azure'
 import { createGoogleGenerativeAI } from '@ai-sdk/google'
 import { createMistral } from '@ai-sdk/mistral'
 import { createOpenAI } from '@ai-sdk/openai'
+import { Message, MessageRole } from '@latitude-data/compiler'
+import { JSONSchema7 } from 'json-schema'
+import { z } from 'zod'
 
-import { Providers } from '../../constants'
+import { Providers, RunErrorCodes } from '../../constants'
+import { Result } from '../../lib'
+import { ChainError } from '../chains/ChainErrors'
+
+const googleCategorySettings = z.union([
+  z.literal('HARM_CATEGORY_HATE_SPEECH'),
+  z.literal('HARM_CATEGORY_DANGEROUS_CONTENT'),
+  z.literal('HARM_CATEGORY_HARASSMENT'),
+  z.literal('HARM_CATEGORY_SEXUALLY_EXPLICIT'),
+])
+const googleThresholdSettings = z.union([
+  z.literal('HARM_BLOCK_THRESHOLD_UNSPECIFIED'),
+  z.literal('BLOCK_LOW_AND_ABOVE'),
+  z.literal('BLOCK_MEDIUM_AND_ABOVE'),
+  z.literal('BLOCK_ONLY_HIGH'),
+  z.literal('BLOCK_NONE'),
+])
+
+export const googleConfig = z
+  .object({
+    structuredOutputs: z.boolean().optional(),
+    cachedContent: z.string().optional(),
+    safetySettings: z
+      .array(
+        z
+          .object({
+            category: googleCategorySettings.optional(),
+            threshold: googleThresholdSettings.optional(),
+          })
+          .optional(),
+      )
+      .optional(),
+  })
+  .optional()
+type GoogleConfig = z.infer<typeof googleConfig>
 
 export type Config = {
-  [key: string]: any
```

Comment:
This was making this type useless. All the `config.tools` and the rest where `any`